### ISSUE TITLE
run_demo.sh partitioning issue again

### ISSUE THREAD

**Issue by eflumerf** [art-daq/artdaq#129]
*This issue has been migrated from https://cdcvs.fnal.gov/redmine/issues/21969 (FNAL account required)*
*Originally created by @ron003 on 2019-02-22 23:14:31*

----

John and I have discovered an issue with the now two different PARTITION_NUMBER env vars. (ARTDAQ_PARTITION_NUMBER and DAQINTRFACE_PARTITION_NUMBER.
This could be considered an extension of issue 20889, but since the associated branch has been merged into develop, it may
be better to create a new issue.

**Comment by eflumerf** [comment]
*Comment by @ron003 on 2019-02-22 23:32:24*

----

At this point changes to both artdaq_demo's run_demo.sh and daqinterface's daqinterface_functions.sh need to happen.
Also, at this point, the changes to run_demo.sh could/should happen first.

**Comment by eflumerf** [comment]
*Comment by @jcfreeman2 on 2019-02-23 19:32:35*

----

To add to what Ron said: we discovered that in run_demo.sh, there's the following snippet in the get_dispatcher_port() function:
<pre>
export DAQINTERFACE_PARTITION_NUMBER=$ARTDAQ_PARTITION_NUMBER
</pre>
where ARTDAQ_PARTITION_NUMBER is set via the "--partition" option to run_demo, but if this option isn't passed, the script will run - but in the line above, the DAQINTERFACE_PARTITION_NUMBER ends up not being an integer anymore. Things can still work given Ron's fix to the internal DAQInterface utility script, artdaq-utilities-daqinterface/bin/daqinterface_functions.sh, in which if DAQINTERFACE_PARTITION_NUMBER has wound up set to "nothing" it defaults to zero:
<pre>
export DAQINTERFACE_PORT=$(( $ARTDAQ_BASE_PORT + ${DAQINTERFACE_PARTITION_NUMBER:-0} * $ARTDAQ_PORTS_PER_PARTITION ))
</pre>
...but it would be preferable to not execute "<code>export DAQINTERFACE_PARTITION_NUMBER=</code> " in run_demo.sh  in the first place.
